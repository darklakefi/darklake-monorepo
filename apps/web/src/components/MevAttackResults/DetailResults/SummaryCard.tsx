import { ReactNode } from "react";

export default function SummaryCard({ title, content }: { title: string, content: ReactNode }) {
  return (
    <div className="flex flex-col bg-brand-60 gap-[16px] p-[16px] h-full">
      <div className="text-body-2 text-brand-30">{title}</div>
      {content}
    </div>
  );
}
