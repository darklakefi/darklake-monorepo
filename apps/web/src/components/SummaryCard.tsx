import { ReactNode } from "react";

export default function SummaryCard({ title, content }: { title: string; content: ReactNode }) {
  return (
    <div className="flex flex-col bg-brand-60 gap-4 p-4 h-full">
      <div className="text-lg leading-6 tracking-normal text-brand-30 uppercase">{title}</div>
      {content}
    </div>
  );
}
