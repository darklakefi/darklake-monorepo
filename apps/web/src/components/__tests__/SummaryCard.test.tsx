import { act, render, RenderResult } from "@testing-library/react";
import SummaryCard from "../SummaryCard";

describe("<SummaryCard />", () => {
  it("displays content successfully", async () => {
    let baseElement: RenderResult["baseElement"] | undefined;

    await act(async () => {
      ({ baseElement } = render(
        <SummaryCard
          title="Total Extracted"
          content={
            <div className="flex flex-col gap-[2px] py-[1px]">
              <div className="text-body text-brand-20">16.32 SOL</div>
              <div className="text-body-2 text-brand-30">2,774.32 USDC</div>
            </div>
          }
        />,
      ));
    });

    expect(baseElement).toMatchSnapshot();
  });

  it("displays content successfully", async () => {
    let baseElement: RenderResult["baseElement"] | undefined;

    await act(async () => {
      ({ baseElement } = render(
        <SummaryCard
          title="Confirmed Attack"
          content={
            <div className="flex flex-col gap-[2px] py-[1px]">
              <div className="text-body text-brand-20">47</div>
            </div>
          }
        />,
      ));
    });

    expect(baseElement).toMatchSnapshot();
  });
});
