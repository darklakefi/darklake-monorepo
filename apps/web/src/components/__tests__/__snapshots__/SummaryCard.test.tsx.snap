// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SummaryCard /> displays content successfully 1`] = `
<body>
  <div>
    <div
      class="flex flex-col bg-brand-60 gap-4 p-4 h-full"
    >
      <div
        class="text-lg leading-6 text-brand-30 uppercase"
      >
        Total Extracted
      </div>
      <div
        class="flex flex-col gap-[2px] py-[1px]"
      >
        <div
          class="text-3xl leading-8 text-brand-20"
        >
          16.32 SOL
        </div>
        <div
          class="text-lg leading-6 text-brand-30"
        >
          2,774.32 USDC
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`<SummaryCard /> displays content successfully 2`] = `
<body>
  <div>
    <div
      class="flex flex-col bg-brand-60 gap-4 p-4 h-full"
    >
      <div
        class="text-lg leading-6 text-brand-30 uppercase"
      >
        Confirmed Attack
      </div>
      <div
        class="flex flex-col gap-[2px] py-[1px]"
      >
        <div
          class="text-3xl leading-8 text-brand-20"
        >
          47
        </div>
      </div>
    </div>
  </div>
</body>
`;
